packages/services/src/inversify.config.ts

```diff
@@ -0,0 +1,152 @@
+import { Container } from 'inversify';
+
+import {
+  DefaultNetworkConfigurations,
+  WalletDefaultNetworkConfigurationIds,
+  defaultNetworksKeyedById,
+} from '@leather.io/models';
+
+import { AlexSdkClient, createAlexSdkClient } from './infrastructure/api/alex-sdk/alex-sdk.client';
+import {
+  BestInSlotApiClient,
+  createBestInSlotApiClient,
+} from './infrastructure/api/best-in-slot/best-in-slot-api.client';
+import {
+  BinanceApiClient,
+  createBinanceApiClient,
+} from './infrastructure/api/binance/binance-api.client';
+import {
+  CoincapApiClient,
+  createCoincapApiClient,
+} from './infrastructure/api/coincap/coincap-api.client';
+import {
+  CoinGeckoApiClient,
+  createCoinGeckoApiClient,
+} from './infrastructure/api/coingecko/coingecko-api.client';
+import { CacheService } from './infrastructure/cache/cache.service';
+import {
+  RateLimiterService,
+  createRateLimiterService,
+} from './infrastructure/rate-limiter/rate-limiter.service';
+import { NetworkSettingsService } from './infrastructure/settings/network-settings.service';
+import { MarketDataService, createMarketDataService } from './market-data/market-data.service';
+
+let currentNetworkContainer: Container;
+type NetworkSpecificContainers = Record<string, Container>;
+const networkSpecificContainers: Partial<NetworkSpecificContainers> = {};
+
+export function initializeServiceContainers(
+  defaultNetworkSettingsService: NetworkSettingsService,
+  cacheService: CacheService
+): Container {
+  if (!currentNetworkContainer) {
+    currentNetworkContainer = new Container();
+    registerDependencies(currentNetworkContainer, defaultNetworkSettingsService, cacheService);
+    for (const networkId of Object.keys(defaultNetworksKeyedById)) {
+      const networkSpecificContainer = new Container();
+      registerDependencies(
+        networkSpecificContainer,
+        {
+          getConfig() {
+            return defaultNetworksKeyedById[networkId as WalletDefaultNetworkConfigurationIds];
+          },
+        },
+        cacheService
+      );
+      networkSpecificContainers[networkId] = networkSpecificContainer;
+    }
```

Comment:
Should we leave this out of the initial implementation? The requirement of looking up the same data of different chains was something worth considering, but not something we need now

Reply:
Agreed, let's YAGNI this out and keep it in our back pockets until we have a concerete use case. Good to know that initialization will be lazy - i.e. the containers register upfront but the actual network-specific services won't instantiate until called for the first time.